tests/drivers/location/src/test_api.c

```diff
@@ -0,0 +1,289 @@
+/*
```

Comment:
Would be good to split the driver code from the use of driver test.  This makes it more like a real driver case, and shows any issues with things like `extern's` that might be needed to access various structs, etc.

Reply:
I know, i should not have included the location API in this draft, although i did get some good feedback for it :)

Reply:
that doesn't concern me, its the need to show how this works for a set of drivers and users.

For example, w/o having a test driver that is standalone we aren't seeing what I think is a N^2 issue with handling the externs to the API structs.